newIDE/app/src/BehaviorsEditor/NewBehaviorDialog.js

```diff
@@ -120,81 +56,80 @@ export default function NewBehaviorDialog({
+    () =>
+      allInstalledBehaviorMetadataList.filter(
+        behavior => !deprecatedBehaviorsInformation[behavior.type]
+      ),
+    [allInstalledBehaviorMetadataList, deprecatedBehaviorsInformation]
+  );
+
+  const deprecatedBehaviorMetadataList: Array<SearchableBehaviorMetadata> = React.useMemo(
     () => {
-      setTimeout(() => {
-        if (shouldAutofocusSearchbar && searchBar.current)
-          searchBar.current.focus();
-      }, 20 /* Be sure that the search bar is shown */);
+      const deprecatedBehaviors = allInstalledBehaviorMetadataList.filter(
+        behavior => deprecatedBehaviorsInformation[behavior.type]
+      );
+      deprecatedBehaviors.forEach(behavior => (behavior.isDeprecated = true));
+      return deprecatedBehaviors;
     },
-    [shouldAutofocusSearchbar]
+    [allInstalledBehaviorMetadataList, deprecatedBehaviorsInformation]
   );
 
   if (!open || !project) return null;
 
-  const deprecatedBehaviorsInformation = getDeprecatedBehaviorsInformation();
-
-  const filteredBehaviorMetadata = filterEnumeratedBehaviorMetadata(
-    behaviorsMetadata,
-    searchText
-  );
-  const behaviors = filteredBehaviorMetadata.filter(
-    ({ type }) => !deprecatedBehaviorsInformation[type]
-  );
-  const deprecatedBehaviors = filteredBehaviorMetadata.filter(
-    ({ type }) => !!deprecatedBehaviorsInformation[type]
-  );
-
-  const _chooseBehavior = (
-    i18n: I18nType,
-    { type, defaultName }: EnumeratedBehaviorMetadata
-  ) => {
-    if (deprecatedBehaviorsInformation[type]) {
-      showMessageBox(i18n._(deprecatedBehaviorsInformation[type].warning));
+  const _chooseBehavior = (i18n: I18nType, behaviorType: string) => {
+    if (deprecatedBehaviorsInformation[behaviorType]) {
+      showMessageBox(
+        i18n._(deprecatedBehaviorsInformation[behaviorType].warning)
+      );
     }
 
-    return onChoose(type, defaultName);
+    const behaviorMetadata = gd.MetadataProvider.getBehaviorMetadata(
+      project.getCurrentPlatform(),
+      behaviorType
+    );
+
+    return onChoose(behaviorType, behaviorMetadata.getDefaultName());
   };
   const chooseBehavior = addCreateBadgePreHookIfNotClaimed(
     authenticatedUser,
     TRIVIAL_FIRST_BEHAVIOR,
     _chooseBehavior
   );
 
-  const isAmongObjectBehaviors = (
-    behaviorMetadata: EnumeratedBehaviorMetadata
-  ) => objectBehaviorsTypes.includes(behaviorMetadata.type);
-
-  const canBehaviorBeUsed = (behaviorMetadata: EnumeratedBehaviorMetadata) => {
-    // An empty object type means the base object, i.e: any object.
-    return (
-      (behaviorMetadata.objectType === '' ||
-        behaviorMetadata.objectType === objectType) &&
-      !isAmongObjectBehaviors(behaviorMetadata)
-    );
-  };
-
-  const hasSearchNoResult =
-    !!searchText && !behaviors.length && !deprecatedBehaviors.length;
-
   const onInstallExtension = async (
     i18n: I18nType,
-    extensionShortHeader: ExtensionShortHeader
+    extensionShortHeader: { url: string }
```

Comment:
Why is this reduced type necessary?

Reply:
When installing an extension from the behavior list, we only have an `BehaviorShortHeader`.